### PR TITLE
Add Module.md files for different artifacts with module and package info

### PR DESCRIPTION
**Pull request by ZacSweers**
This is what dokka uses to document packages and modules. Deleted the `package-info.kt` file as it doesn't do anything on its own, but can restore if it's still wanted

<kbd>
  
</kbd>

<kbd>
  
</kbd>

### CODE REVIEW

File: okhttp/Module.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Module okhttp
+
+An HTTP+HTTP/2 client for Android and Java applications.
+
+# Package okhttp3
+
+An HTTP+HTTP/2 client for Android and Java applications.
```
Comment (swankjesse) [inline okhttp/Module.md]: What's the utility in repeating these sentences? We should pick one.

Comment (ZacSweers) [reply okhttp/Module.md]: One's for the module, and you can have any number of packages with their own. Basically the package lines are the replacement for package-info documentation, and the module line is per-module documentation (either a blurb or you could make it fully filled like an embedded README)

Example of multiple packages

```markdown
# Module okhttp

An HTTP+HTTP/2 client for Android and Java applications.

# Package okhttp3

An HTTP+HTTP/2 client for Android and Java applications.

# Package okhttp3.hypothetical

A hypothetical other package doc.

```

<kbd>
  
</kbd>

Comment (swankjesse) [reply okhttp/Module.md]: Can we defer that until we actually have a module with multiple packages?

Comment (ZacSweers) [reply okhttp/Module.md]: That works. Then it looks like this

<kbd>
  
</kbd>

Comment (ZacSweers) [reply okhttp/Module.md]: Or alternatively you could have the package only and not the module. I did modules only in 3dc147c but should have asked first. Lmk which you prefer

<kbd>
  
</kbd>

**Pull request merged**